core/src/main/scala/chisel3/experimental/package.scala

```diff
@@ -130,6 +132,35 @@ package object experimental {
     }
   }
 
+  /** This class provides the `Lit` method needed to define a `Vec` literal
+    */
+  object VecLiterals {
+    implicit class AddVecLiteralConstructor[T <: Data](x: Vec[T]) {
+      /** Given a generator of a list tuples of the form [Int, Data]
+        * constructs a Vec literal, parallel concept to `BundleLiteral`
+        *
+        * @param elems tuples of an index and a literal value
+        * @return
+        */
+      def Lit(elems: (Int, T)*): Vec[T] = {
+        x._makeLit(elems: _*)
+      }
+    }
+
+    implicit class AddObjectLiteralConstructor(x: Vec.type) {
+      /** This provides an literal construction method for cases using
+        * object `Vec` as in `Vec.Lit(1.U, 2.U)`
+        */
+      def Lit[T <: Data](elems: T*)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Vec[T] = {
```

Comment:
```suggestion
      def Lit[T <: Data](elems: T*): Vec[T] = {
```

I think we should remove these implicits for now. We need a `SourceInfo` macro to handle varargs so that one can do a bit extraction immediately after the `Lit` method. However, given that we're lacking the implicits from BundleLiterals as well, I think all of these `Lit` methods should be consistent.

Reply:
Made this change.
@ducky64 @jackkoenig let me know if anything else needs to be done